iodata/formats/pdb.py

```diff
@@ -0,0 +1,132 @@
+from ..utils import angstrom, LineIterator
+
+
+__all__ = []
+
+
+PATTERNS = ['*.pdb']
+
+
+@document_load_one("PDB", ['atcoords', 'atnums'], ['title'])
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    nums = []
+    coords = []
+    attypes = []
+    restypes = []
+    molecule_found = False
+    while True:
+        try:
+            line = next(lit)
+        except StopIteration:
+            break
+        if line.startswith("TITLE") or line.startswith("COMPND"):
+            words = line.split()
+            title = words[1:]
+        else:
+            title = "Pdb file from Iodata"
+        if line[0:4] == "ATOM" or line[0:6] == "HETATM":
+            try:
+                # Try reading element from position 77:79 referenced in pdb format
+                words = line[77:79].split()
+                if not words:
+                    # If not guess it from the atom type
+                    words = line[13:16].split()
+                symbol = words[0].title()
+                nums.append(sym2num.get(symbol, sym2num.get(symbol[0], None)))
+            except ValueError:
+                # When the position 77:79 does not exist also use the atom type
+                symbol = line[13:16].title()
+                nums.append(sym2num.get(symbol, sym2num.get(symbol[0], None)))
+            x = float(line[30:38])
+            y = float(line[38:46])
+            z = float(line[46:54])
+            coords.append([x, y, z])
+            attypes.append(line[13:16].strip())
+            restypes.append(line[18:20].strip())
+            molecule_found = True
+        if line[0:3] == "END" and molecule_found:
+            atnums = np.array(nums)
+            atcoords = np.array(coords) * angstrom
+            attypes = tuple(attypes)
+            restypes = tuple(restypes)
+            atffparams = {"attypes": attypes, "restypes": restypes}
+            result = {
+                'atcoords': atcoords,
+                'atnums': atnums,
+                'atffparams': atffparams,
+                'title': title
+            }
+            break
+    if molecule_found is False:
+        raise lit.error("Molecule could not be read")
+    return result
+
+
+@document_load_many("PDB", ['atcoords', 'atnums', 'atffparams'], ['title'])
+def load_many(lit: LineIterator) -> Iterator[dict]:
+    """Do not edit this docstring. It will be overwritten."""
+    # PDB files with more molecules are a simple concatenation of individual MOL2 files,'
+    # making it trivial to load many frames.
+    while True:
+        try:
+            yield load_one(lit)
+        except IOError:
+            return
+
+
+@document_dump_one("PDB", ['atcoords', 'atnums'], ['atffparams', 'title'])
+def dump_one(f: TextIO, data: IOData):
+    """Do not edit this docstring. It will be overwritten."""
+    print(str("TITLE    " + data.title) or "TITLE    Created with IOdata", file=f)
+    attypes = data.atffparams.get('attypes')
```

Comment:
This would generate a `ValueError` when `attypes` is not present. Suggested fix:
```suggestion
    attypes = data.atffparams.get('attypes', None)
```
To test this case, you could add a unit test that loads an XYZ file and writes it back out as PDB file to a temporary directory. You could even load the PDB again to check if it has the expected contents.
(Same comment for next line.)

Reply:
I added a test for the attype and restype read and write from an xyz file.

Reply:
looks good!